src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -82,6 +82,53 @@ by simp [←coe_int_mul_eq_zsmul]
 @[simp] lemma coe_pi_add_coe_pi : (π : real.angle) + π = 0 :=
 by rw [←two_nsmul, two_nsmul_coe_pi]
 
+lemma two_nsmul_eq_iff {ψ θ : angle} : (2 : ℕ) • ψ = (2 : ℕ) • θ ↔ (ψ = θ ∨ ψ = θ + π) :=
+begin
+  induction ψ using real.angle.induction_on,
+  induction θ using real.angle.induction_on,
+  simp_rw [←coe_nat_mul_eq_nsmul],
```

Comment:
Does adding these two lemmas help at all?
```
@[simp] lemma coe_nsmul (n : ℕ) (x : ℝ) : ↑(n • x : ℝ) = (n • ↑x : angle) := rfl
@[simp] lemma coe_zsmul (z : ℤ) (x : ℝ) : ↑(z • x : ℝ) = (z • ↑x : angle) := rfl
```
(we should probably have them anyway)

Reply:
I don't think those lemmas are particularly relevant for this PR (in that `nsmul` and `zsmul` on `ℝ` don't seem particularly useful for this PR), but they seem reasonable enough for adding separately.

Reply:
#15003